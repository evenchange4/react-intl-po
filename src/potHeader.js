import R from 'ramda';

/**
 * Create a POT header string
 * @param {Object} options
 * @param {String|String[]} [options.comments]
 * @param {Date} [options.potCreationDate] used for POT-Creation-Date
 * @param {String} [options.projectIdVersion] Project-Id-Version
 * @param {String} [options.charset]
 * @param {String} [options.encoding]
 * @return {String} potSource
 *
 * example: see tests
 *
 * @see https://www.gnu.org/software/trans-coord/manual/gnun/html_node/PO-Header.html
 * @author Guillaume Boddaert
 */

const potHeader = (options = {}) => {
  const o = R.evolve({
    comments: R.pipe(
      R.cond([[R.is(Array), R.identity], [R.is(String), R.of]]),
      R.map(R.split('\n')),
      R.flatten,
      R.map(e => `# ${e}`),
      R.join('\n'),
    ),
    projectIdVersion: e => `"Project-Id-Version: ${e}\\n"`,
    potCreationDate: e => `"POT-Creation-Date: ${e.toISOString()}\\n"`,
    charset: e => `"Content-Type: text/plain; charset=${e}\\n"`,
    encoding: e => `"Content-Transfer-Encoding: ${e}\\n"`,
  })(options);

  return `${o.comments}
msgid ""
msgstr ""
${o.projectIdVersion}
${o.potCreationDate}
${o.charset}
${o.encoding}
"MIME-Version: 1.0\\n"
"X-Generator: react-intl-po\\n"


`.replace(/undefined\r?\n|\r/g, '');
};

export default potHeader;
