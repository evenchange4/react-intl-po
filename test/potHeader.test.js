import potHeader from '../src/potHeader';

it('should return a function', () => {
  expect(typeof potHeader).toBe('function');
});

it('should return pot header, without any parameter', () => {
  expect(potHeader()).toMatchSnapshot();
});

it('should return pot header, without empty options', () => {
  expect(potHeader({})).toMatchSnapshot();
});

it('should return pot header, with a single comment', () => {
  expect(
    potHeader({
      comments: 'This is a single line comment',
    }),
  ).toMatchSnapshot();
});

it('should return pot header, with a single comment, with CRÂ in it', () => {
  expect(
    potHeader({
      comments: 'This is a multi-line\ncomment\n',
    }),
  ).toMatchSnapshot();
});

it('should return pot header, with a list of comments', () => {
  expect(
    potHeader({
      comments: ['A', 'B', 'C'],
    }),
  ).toMatchSnapshot();
});

it('should return pot header, with all options', () => {
  expect(
    potHeader({
      comments: 'This is a single line comment',
      projectIdVersion: 'FUBAR',
      potCreationDate: new Date(Date.UTC(1995, 11, 17, 3, 24, 0)),
      charset: 'UTF-8',
      encoding: '8bit',
    }),
  ).toMatchSnapshot();
});
