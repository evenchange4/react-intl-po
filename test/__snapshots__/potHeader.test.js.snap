// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return pot header, with a list of comments 1`] = `
# A
# B
# C
msgid ""
msgstr ""
"MIME-Version: 1.0\\n"
"X-Generator: react-intl-po\\n"



`;

exports[`should return pot header, with a single comment 1`] = `
# This is a single line comment
msgid ""
msgstr ""
"MIME-Version: 1.0\\n"
"X-Generator: react-intl-po\\n"



`;

exports[`should return pot header, with a single comment, with CRÂ in it 1`] = `
# This is a multi-line
# comment
# 
msgid ""
msgstr ""
"MIME-Version: 1.0\\n"
"X-Generator: react-intl-po\\n"



`;

exports[`should return pot header, with all options 1`] = `
# This is a single line comment
msgid ""
msgstr ""
"Project-Id-Version: FUBAR\\n"
"POT-Creation-Date: 1995-12-17T03:24:00.000Z\\n"
"Content-Type: text/plain; charset=UTF-8\\n"
"Content-Transfer-Encoding: 8bit\\n"
"MIME-Version: 1.0\\n"
"X-Generator: react-intl-po\\n"



`;

exports[`should return pot header, without any parameter 1`] = `
msgid ""
msgstr ""
"MIME-Version: 1.0\\n"
"X-Generator: react-intl-po\\n"



`;

exports[`should return pot header, without empty options 1`] = `
msgid ""
msgstr ""
"MIME-Version: 1.0\\n"
"X-Generator: react-intl-po\\n"



`;
